graylog2-web-interface/src/views/components/searchbar/queryvalidation/QueryValidationContext.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+import { createContext, useContext } from 'react';
+
+export const QueryValidationContext = createContext(undefined);
+export const QueryValidationProvider = QueryValidationContext.Provider;
+
+export const useQueryValidationContext = <T>() => {
+  return useContext<T>(QueryValidationContext);
```

Comment:
Since we only use this context in a plugin we should not store in in the core repo.